syft/pkg/cataloger/golang/licenses.go

```diff
@@ -138,3 +163,79 @@ func processCaps(s string) string {
 		return "!" + strings.ToLower(s)
 	})
 }
+
+func getModule(progress *event.GenericProgress, proxies []string, moduleName, moduleVersion string) (fsys fs.FS, err error) {
+	for _, proxy := range proxies {
+		u, _ := url.Parse(proxy)
+		if proxy == "direct" {
+			fsys, err = getModuleRepository(progress, moduleName, moduleVersion)
+			continue
+		}
+		switch u.Scheme {
+		case "https", "http":
+			fsys, err = getModuleProxy(progress, proxy, moduleName, moduleVersion)
+		case "file":
+			p := filepath.Join(u.Path, moduleName, "@v", moduleVersion)
+			progress.SetValue(fmt.Sprintf("file: %s", p))
+			fsys = os.DirFS(p)
+		}
+		if fsys != nil {
+			break
+		}
+	}
+	return
+}
+
+//nolint:gosec
```

Comment:
only nolint the line that is failing, not the entire function (probably on `http.Get`)
```
	resp, err := http.Get(u)  //nolint:gosec
```